src/faultlog/faultlog_main.cpp

```diff
@@ -74,6 +75,60 @@ void createNagPel(sdbusplus::bus::bus& bus,
     }
 }
 
+/** @brief Callback method for boot progress property change
+ *
+ *  @param[in] bus - D-Bus to attach to
+ *  @param[in] unresolvedRecords - hardware isolated records to parse
+ *  @param[in] msg - property change D-Bus message
+ */
+void propertyChanged(sdbusplus::bus::bus& bus,
+                     const GuardRecords& unresolvedRecords,
+                     sdbusplus::message::message& msg)
+{
+    using ProgressStages = sdbusplus::xyz::openbmc_project::State::Boot::
+        server::Progress::ProgressStages;
+    using PropertiesVariant =
+        sdbusplus::utility::dedup_variant_t<ProgressStages>;
+    using Properties = std::map<std::string, PropertiesVariant>;
+    std::string intf;
+    Properties propMap;
+    msg.read(intf, propMap);
+    for (const auto& [prop, propValue] : propMap)
+    {
+        if (prop == "BootProgress")
+        {
+            const ProgressStages* progPtr =
+                std::get_if<ProgressStages>(&propValue);
+            if (progPtr != nullptr)
+            {
+                lg2::info("faultlog - host poweron check boot progress "
+                          "value is "
+                          "{BOOT_PROGRESS}",
+                          "BOOT_PROGRESS", *progPtr);
+                if ((*progPtr == ProgressStages::MemoryInit) ||
+                    (*progPtr == ProgressStages::SecondaryProcInit) ||
+                    (*progPtr == ProgressStages::PCIInit) ||
+                    (*progPtr == ProgressStages::SystemInitComplete) ||
+                    (*progPtr == ProgressStages::SystemSetup) ||
+                    (*progPtr == ProgressStages::OSStart) ||
```

Comment:
can we remove memory init, secondary proc init PCI init from this check, let us wait till PHYP start to start the NAG

Reply:
ok